### PR TITLE
Review option

### PR DESCRIPTION
**Pull request by srikar-1997**
Please review my work.

### CODE REVIEW

File: project1/application.py [hunk 1]
```diff
@@ -82,3 +83,52 @@ def admin():
...
+    if request.method == "GET":
+        isb = Review.query.filter_by(ISBN_No = ISBN).all()
+        rev = Review.query.filter(and_(Review.name == name, Review.ISBN_No == ISBN)).first()
+        if rev and rev.flag == 2 :
+            return render_template("submit.html", sub_flag = 1, rate = rev.review_rate, comm = rev.review_description, isb = isb)
+        return render_template("submit.html", isb = isb, ISBN = ISBN)
+
+    
+
+@app.route("/book/<ISBN>", methods = ["POST"])
```
Comment (rishikaharithaMSIT) [inline project1/application.py]: Are you using same route for book page and review function?

File: project1/models.py [hunk 2]
```diff
@@ -24,4 +24,20 @@ def __init__ (self, ISBN, title, author, year) :
...
+
+class Review(db.Model) :
+    __tablename__ = "reviews"
+
+    __table_args__ = (db.PrimaryKeyConstraint('name', 'ISBN_No'),)
+    name = db.Column(db.String, db.ForeignKey('users.username'))
+    ISBN_No = db.Column(db.String, db.ForeignKey('books.ISBN'))
+    review_rate = db.Column(db.Integer)
+    review_description = db.Column(db.String)
+    flag = db.Column(db.Integer, default = 0)
```
Comment (rishikaharithaMSIT) [inline project1/models.py]: What does this flag do?

File: project1/application.py [hunk 3]
```diff
@@ -82,3 +83,52 @@ def admin():
...
 
+@app.route("/book/<ISBN>", methods = ["GET"])
+def book_details(ISBN):
+    name = session["name"]
+    # ISBN = "1416949658"
+    if request.method == "GET":
+        isb = Review.query.filter_by(ISBN_No = ISBN).all()
+        rev = Review.query.filter(and_(Review.name == name, Review.ISBN_No == ISBN)).first()
+        if rev and rev.flag == 2 :
+            return render_template("submit.html", sub_flag = 1, rate = rev.review_rate, comm = rev.review_description, isb = isb)
```
Comment (rishikaharithaMSIT) [inline project1/application.py]: Is submit.html the page in which the reviews will be available?

**Review by rishikaharithaMSIT — CHANGES_REQUESTED** [review]

File: project1/application.py (hunk 1 above)
Comment (srikar-1997) [reply project1/application.py]: yeah as of now we were using same route but we have to check after integrating

File: project1/models.py (hunk 2 above)
Comment (srikar-1997) [reply project1/models.py]: Removed.

File: project1/application.py (hunk 3 above)
Comment (srikar-1997) [reply project1/application.py]: changes the name to bookpage.html

Comment (srikar-1997) [reply project1/application.py]: changed*

**Review by rishikaharithaMSIT — APPROVED** [review]

**Pull request merged**